internal/core/analysis_api/handlers/github.go

```diff
@@ -0,0 +1,330 @@
+			default:
+				zap.L().Debug("skipped unsupported file", zap.String("fileName", zipFile.Name))
+			}
+
+			if err != nil {
+				return nil, nil, err
+			}
+
+			// Map the Config struct fields over to the fields we need to store in Dynamo
+			analysisItem := tableItemFromConfig(config)
+			if analysisItem.Type == models.TypeDataModel {
+				// ensure Mappings are nil rather than an empty slice
+				if len(config.Mappings) > 0 {
+					analysisItem.Mappings = make([]models.DataModelMapping, len(config.Mappings))
+					for i, mapping := range config.Mappings {
+						analysisItem.Mappings[i], err = buildMapping(mapping)
+						if err != nil {
+							return nil, nil, err
+						}
+					}
+				}
+			}
+
+			for i, test := range config.Tests {
+				// A test can specify a resource and a resource type or a log and a log type.
+				// By convention, log and log type are used for rules and resource and resource type are used for policies.
+				if test.Resource == nil {
+					analysisItem.Tests[i], err = buildRuleTest(test)
+				} else {
+					analysisItem.Tests[i], err = buildPolicyTest(test)
+				}
+				if err != nil {
+					return nil, nil, err
+				}
+			}
+
+			if _, exists := detections[analysisItem.ID]; exists {
+				return nil, nil, fmt.Errorf("multiple analysis specs with ID %s", analysisItem.ID)
+			}
+			detections[analysisItem.ID] = analysisItem
+		}
+	}
+
+	// add python bodies
+	// Finish each policy by adding its body and then validate it
+	for _, detection := range detections {
+		if body, ok := detectionBodies[detection.Body]; ok {
+			detection.Body = body
+			if err := validateUploadedPolicy(detection); err != nil {
+				return nil, nil, err
+			}
+		} else if detection.Type != models.TypeDataModel {
+			// it is ok for DataModels to be missing python body
+			return nil, nil, fmt.Errorf("policy %s is missing a body", detection.ID)
+		}
+	}
+
+	return packs, detections, err
+}
+
+func listAvailableGithubReleases() ([]models.Version, error) {
+	// Setup options and client
+	// TODO: should number of available releases be configurable? By default returns all releases
+	// paged at 100 releases at a time
+	opt := &github.ListOptions{}
+	var allReleases []*github.RepositoryRelease
+	for {
+		releases, response, err := githubClient.Repositories.ListReleases(context.Background(), githubOwner, githubRepo, opt)
+		if err != nil {
+			return nil, err
+		}
+		allReleases = append(allReleases, releases...)
+		if response.NextPage == 0 {
+			break
+		}
+		opt.Page = response.NextPage
+	}
+	var availableVersions []models.Version
+	// earliest version of panther managed detections that supports packs
+	minimumVersion, _ := version.NewVersion(minimumVersionName)
+	for _, release := range allReleases {
+		version, err := version.NewVersion(*release.Name)
+		if err != nil {
+			// if we can't parse the version, just throw it away?
+			zap.L().Warn("can't parse version", zap.String("version", *release.Name))
```

Comment:
Yeah this seems correct
```suggestion
			// if we can't parse the version, just throw it away
			zap.L().Warn("can't parse version", zap.String("version", *release.Name))
```